aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -121,15 +122,15 @@ Tensor& linalg_det_out(const Tensor& self, Tensor& out) {
 }
 
 Tensor linalg_det(const Tensor& self) {
-  squareCheckInputs(self);
+  squareCheckInputs(self, "linalg_det");
   TORCH_CHECK((at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type())),
               "Expected a floating point or complex tensor as input");
```

Comment:
I think these checks could be gathered into a single function again with a string argument for the function that is being checked.

Reply:
Done. Again, we should do a decent clean-up of linalg for 1.11, now that the functionality is sort of complete. I'm sure that while doing that we'll be able to make the code much easier to maintain and will really payoff in the future.